### ISSUE TITLE
Please update the documentation

### ISSUE THREAD

**Issue by piegamesde** [MGlolenstine/gtk4-cross#10]
I am trying to get this to work, but I'm really struggling. (Also, this is my first time using Docker so I may be dumber than your average user)

- The command for the base image, `docker build gtk4-cross-base -t gtk4-cross-base` does not work at all. I believe this is because the correct name is now `gtk4-cross-base-hub`
- Even when correcting the path, #4 prevents the image from building without further interaction
- If there pre-built images that make manually building them optional, this would be cool to note too. Pulling from Docker hub with `docker pull mglolenstine/gtk4-cross` did not work ("manifest for mglolenstine/gtk4-cross:latest not found: manifest unknown: manifest unknown"), but `docker pull ghcr.io/mglolenstine/gtk4-cross:rust-gtk-4.4`

-----

I finally got things to work, the only missing step is that my application fails to link after compiling: 

```
          /usr/bin/ld: cannot find -lpangowin32-1.0
          /usr/bin/ld: cannot find -lpangocairo-1.0
          /usr/bin/ld: cannot find -lpango-1.0
          /usr/bin/ld: cannot find -lharfbuzz
          /usr/bin/ld: cannot find -lgdk_pixbuf-2.0
          /usr/bin/ld: cannot find -lcairo-gobject
          /usr/bin/ld: cannot find -lcairo
          /usr/bin/ld: cannot find -lgraphene-1.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lgtk-4
          /usr/bin/ld: cannot find -lpangowin32-1.0
          /usr/bin/ld: cannot find -lpangocairo-1.0
          /usr/bin/ld: cannot find -lpango-1.0
          /usr/bin/ld: cannot find -lharfbuzz
          /usr/bin/ld: cannot find -lgdk_pixbuf-2.0
          /usr/bin/ld: cannot find -lcairo-gobject
          /usr/bin/ld: cannot find -lcairo
          /usr/bin/ld: cannot find -lgraphene-1.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lgraphene-1.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lgtk-4
          /usr/bin/ld: cannot find -lpangowin32-1.0
          /usr/bin/ld: cannot find -lpangocairo-1.0
          /usr/bin/ld: cannot find -lpango-1.0
          /usr/bin/ld: cannot find -lharfbuzz
          /usr/bin/ld: cannot find -lgdk_pixbuf-2.0
          /usr/bin/ld: cannot find -lcairo-gobject
          /usr/bin/ld: cannot find -lcairo
          /usr/bin/ld: cannot find -lgraphene-1.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lgdk_pixbuf-2.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lpango-1.0
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lharfbuzz
          /usr/bin/ld: cannot find -lcairo-gobject
          /usr/bin/ld: cannot find -lcairo
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lintl
          /usr/bin/ld: cannot find -lintl
          collect2: error: ld returned 1 exit status
```

I don't know why it can't find those, I tried installing cairo and it said that it was already installed.

**Comment by MGlolenstine** [comment]
Thanks for the report, the documentation really is lacking and I was thinking of improving it.
Thanks for the details, I'll look into it later today.

**Comment by piegamesde** [comment]
I tried switching to rootless podman and now run into weird file system issues, not sure if related

```
STEP 12/15: RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
info: downloading installer
info: profile set to 'default'
info: default host triple is x86_64-unknown-linux-gnu
warning: Updating existing toolchain, profile choice will be ignored
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: latest update on 2022-04-07, rust version 1.60.0 (7737e0b5c 2022-04-04)
info: downloading component 'rust-std' for 'x86_64-pc-windows-gnu'
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: removing previous version of component 'rust-std' for 'x86_64-pc-windows-gnu'
info: rolling back changes
error: could not rename component file from '/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/self-contained' to '/root/.rustup/tmp/wh9cc98konfm1ose_dir/bk': Invalid cross-device link (os error 18)
Error: error building at STEP "RUN curl https://sh.rustup.rs -sSf | sh -s -- -y": error while running runtime: exit status 1
```

**Comment by piegamesde** [comment]
Also, all shell scripts should start with `set -euo pipefail` and pass https://shellcheck.net/

**Comment by MGlolenstine** [comment]
I have added some improvements and pushed them to the new branch `documentation_script_rewrite` and can be seen in #11 .

Could you test it out and give feedback?

**Comment by MGlolenstine** [comment]
Should be fixed by #11 .
